### PR TITLE
Allow exclusion on Delete After Read per category

### PR DESCRIPTION
**Pull request by tobinstultiens**
Created a settings that allows you to set certain categories to ignore delete after read.
Related to #5680 and #4121

### CODE REVIEW

File: app/src/main/java/eu/kanade/tachiyomi/ui/reader/ReaderPresenter.kt [hunk 1]
```diff
@@ -409,8 +409,20 @@ class ReaderPresenter(
         val currentChapterPosition = chapterList.indexOf(currentChapter)
         val removeAfterReadSlots = preferences.removeAfterReadSlots()
         val chapterToDelete = chapterList.getOrNull(currentChapterPosition - removeAfterReadSlots)
+
+        // Retrieve the categories that are set to exclude from being deleted on read
```
Comment (arkon) [inline app/src/main/java/eu/kanade/tachiyomi/ui/reader/ReaderPresenter.kt:413]: I think it'd make more sense to put this in `DownloadManager`.

**Review by arkon — COMMENTED** [review]
Shouldn't this also apply for users manually marking chapters as read too?

File: app/src/main/java/eu/kanade/tachiyomi/ui/reader/ReaderPresenter.kt (hunk 1 above)
Comment (tobinstultiens) [reply app/src/main/java/eu/kanade/tachiyomi/ui/reader/ReaderPresenter.kt]: The download manager was the exact location I was looking for.
I have now placed it there although I needed to add an injection of the db and by doing this it now also works for the marked as read.

**Comment by tobinstultiens** [comment]
Now it also ignores it when it is manually marked as read.

**Comment by arkon** [comment]
Thanks!

**Pull request merged**